writeto.go

```diff
@@ -77,8 +77,13 @@ type messageWriter struct {
 	err        error
 }
 
-func (w *messageWriter) openMultipart(mimeType string) {
+func (w *messageWriter) openMultipart(mimeType, boundary string) {
 	mw := multipart.NewWriter(w)
+	if boundary == "" {
+		boundary = mw.Boundary()
```

Comment:
Correct me if I'm wrong, it doesn't look like there's any need for this assignment unless we decide to use it on L87 in place of `mw.Boundary()`.

Reply:
yes, thanks